### PR TITLE
set data atrribute

### LINKED ISSUE #137: Add a data attribute to notebooks/consoles that support debugging.
We should use a mechanism like this to make sure that classes we add (for example to support gutter hovers, etc.) are only visible in notebooks and consoles where they are relevant. We could then target them via CSS selectors, e.g.:

```css
[data-jp-debugger='true'] .CodeMirror-gutter-wrapper::after {
  content: '●';
  color: var(--jp-error-color1);
  /* ... */
}
```

### PR DESCRIPTION
**Pull request by KsavinN**
resolved #137

### CODE REVIEW

File: src/editors/index.ts [hunk 1]
```diff
@@ -42,6 +42,25 @@ export class DebuggerEditors extends TabPanel {
     this.editorFactory = options.editorServices.factoryService.newInlineEditor;
     this.mimeTypeService = options.editorServices.mimeTypeService;
 
+    this.debuggerService.session?.clientChanged.connect((_, updated) => {
+      if (this.tabBar?.currentTitle?.owner?.id === updated.path) {
```
Comment (jtpio) [inline src/editors/index.ts]: We probably don't need to use the optional chaining with `?` everywhere.

For example here `this.tabBar` should not be `null`.

File: src/session.ts [hunk 2]
```diff
@@ -64,9 +64,14 @@ export class DebugSession implements IDebugger.ISession {
 
     if (client) {
       this._client.iopubMessage.connect(this._handleEvent, this);
+      this._clientChanged.emit(client as IClientSession);
```
Comment (jtpio) [inline src/session.ts]: This PR would need to be rebased to take into account the changes from #231, which adds support for `Session.ISession`.

File: src/editors/index.ts (hunk 1 above)
Comment (KsavinN) [reply src/editors/index.ts]: tabBar exist before. I will just check if currentTitle exist.

File: src/session.ts (hunk 2 above)
Comment (KsavinN) [reply src/session.ts]: resolved.

File: src/editors/index.ts (hunk 1 above)
Comment (KsavinN) [reply src/editors/index.ts]: for session i put optional chaining cause test are not passed

File: src/editors/index.ts [hunk 3]
```diff
@@ -134,6 +152,7 @@ export class DebuggerEditors extends TabPanel {
     editor.title.label = path;
     editor.title.caption = path;
     editor.title.closable = true;
+    editor.id = this.debuggerService.session.client.path;
```
Comment (jtpio) [inline src/editors/index.ts]: Wondering whether we need an `id`, or we could use `editor.title.label`?

File: src/handlers/console.ts [hunk 4]
```diff
@@ -20,6 +20,7 @@ export class ConsoleHandler implements IDisposable {
     this.debuggerModel = options.debuggerService.model as Debugger.Model;
     this.debuggerService = options.debuggerService;
     this.consolePanel = options.widget;
+    this.consolePanel.node.setAttribute('data-jp-debugger', 'true');
```
Comment (jtpio) [inline src/handlers/console.ts]: Should we set the attributes in both the notebook / console handlers and in the editor handler? Or would one place be enough?

File: src/editors/index.ts (hunk 1 above)
Comment (KsavinN) [reply src/editors/index.ts]: Due to we want always to have the possibility to set breakpoints even for not focused editors I removed a connection to clienteChanged and tabBar changed.

Comment (jtpio) [reply src/editors/index.ts]: Then do we need to keep the `clientChanged` signal?

Comment (KsavinN) [reply src/editors/index.ts]: I removed that

Comment (jtpio) [reply src/editors/index.ts]: :+1:

File: src/editors/index.ts (hunk 3 above)
Comment (jtpio) [reply src/editors/index.ts]: After the rebase it looks like it can actually be removed.

Comment (KsavinN) [reply src/editors/index.ts]: Removed.

Comment (jtpio) [reply src/editors/index.ts]: :+1:

**Pull request merged**